packages/manager/apps/web/client/app/hosting/database/private-sql-activation/private-sql-activation.routing.js

```diff
@@ -31,12 +31,36 @@ export default /* @ngInject */ ($stateProvider) => {
           ),
         versions: /* @ngInject */ (PrivateDatabase) =>
           PrivateDatabase.getOrderableDatabaseVersions('classic'),
-        services: /* @ngInject */ (OvhApiHostingWeb) =>
-          OvhApiHostingWeb.v6()
-            .query()
-            .$promise.then((services) =>
-              orderBy(services, (serviceName) => toLower(serviceName)),
+        datacenter: /* @ngInject */ (serviceName, getDatacenter) =>
+          getDatacenter(serviceName),
+        getDatacenter: /* @ngInject */ (Hosting) => async (serviceName) => {
+          const { datacenter } = await Hosting.getHosting(serviceName);
```

Comment:
**question**: are you sure this is AngularJS friendly ? I mean, no issues with the digest cycle or the dirty checking ?

Reply:
Hello, I took this cope snippet from https://github.com/ovh/manager/blob/master/packages/manager/apps/web/client/app/hosting/database/order/private/hosting-database-order-private.routing.js